src/components/vr_cooperation/src/command/base_json_request.cc

```diff
@@ -131,6 +131,7 @@ void BaseJsonRequest::SendMessageToHMI(
     const vr_hmi_api::ServiceMessage& message) {
   LOG4CXX_AUTO_TRACE(logger_);
   parent_->SendMessageToHMI(message);
+  parent_->UnregisterRequest(message.correlation_id());
```

Comment:
I'm not sure that unregistering request should be here because this action deletes the command and you never do receive response.